dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ChatCompletionStreamingResult.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.ChatCompletion;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.AzureSdk;
+
+internal sealed class ChatStreamingResult : IChatStreamingResult, ITextCompletionStreamingResult
+{
+    private readonly StreamingChatChoice _choice;
+
+    public ChatStreamingResult(StreamingChatChoice choice)
+    {
+        this._choice = choice;
+    }
+
+    /// <inheritdoc/>
+    public async Task<IChatMessage> GetChatMessageAsync(CancellationToken cancellationToken = default)
+    {
+        ChatMessage? chatMessage = null;
+        await foreach (var message in this._choice.GetMessageStreaming(cancellationToken))
+        {
+            // As it loops thru the stream, the return ChatMessage is an updated/concatenated version of the previous ChatMessage
+            chatMessage = message;
+        }
+        // As it leaves the streaming loop, the return ChatMessage is the final version of the ChatMessage to present in a non-streaming behavior
+
+        if (chatMessage is null)
+        {
+            throw new AIException(AIException.ErrorCodes.UnknownError);
```

Comment:
Consider adding a few details describing what happened to simplify debugging/troubleshooting process.

Reply:
Good point, added more details to the exception message.